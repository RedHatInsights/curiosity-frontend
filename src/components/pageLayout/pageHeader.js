import React from 'react';
import PropTypes from 'prop-types';
import {
  PageHeader as RcsPageHeader,
  PageHeaderTitle
} from '@redhat-cloud-services/frontend-components/components/PageHeader';

/**
 * Render a platform page header.
 *
 * @param {object} props
 * @param {Node} props.children
 * @returns {Node}
 */
const PageHeader = ({ children }) => (
  <RcsPageHeader>
    <PageHeaderTitle title={children} />
  </RcsPageHeader>
);

/**
 * Prop types.
 *
 * @type {{children: Node}}
 */
PageHeader.propTypes = {
  children: PropTypes.node.isRequired
};

/**
 * Default props.
 */
PageHeader.defaultProps = {};

export { PageHeader as default, PageHeader };
