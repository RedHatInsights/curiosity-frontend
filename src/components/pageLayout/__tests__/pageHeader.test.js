import React from 'react';
import { mount } from 'enzyme';
import { PageHeader } from '../pageHeader';

describe('PageHeader Component', () => {
  it('should render a basic component', () => {
    const component = mount(
      <PageHeader>
        <span className="test">lorem</span>
      </PageHeader>
    );

    expect(component).toMatchSnapshot('basic');
  });

  it('should render string node types', () => {
    const component = mount(<PageHeader>dolor sit</PageHeader>);
    expect(component).toMatchSnapshot('string');
  });

  it('should render the subtitle when viewId is provided', () => {
    const component = mount(<PageHeader productLabel="RHEL">lorem</PageHeader>);
    expect(component).toMatchSnapshot('with subtitle');
  });

  it('should render the tour button when includeTour is provided', () => {
    const component = mount(<PageHeader includeTour>lorem</PageHeader>);
    expect(component).toMatchSnapshot('with tour button');
    expect(component.render()).toMatchSnapshot('consistent html output for Pendo CSS selector');
  });
});
