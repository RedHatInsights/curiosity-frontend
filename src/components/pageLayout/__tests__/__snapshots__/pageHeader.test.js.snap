// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader Component should render a basic component: basic 1`] = `
<PageHeader>
  <Flex
    justifyContent={
      {
        "sm": "justifyContentSpaceBetween",
      }
    }
  >
    <FlexItem>
      <PageHeaderTitle
        className="pf-u-mb-sm"
        title={
          <span
            className="test"
          >
            lorem
          </span>
        }
      />
    </FlexItem>
  </Flex>
</PageHeader>
`;

exports[`PageHeader Component should render string node types: string 1`] = `
<pageheader
  props="{
  "children": [
    {}
  ],
  "productLabel": null
}"
>
  <section
    class="pf-v5-l-page-header pf-v5-c-page-header pf-v5-l-page__main-section pf-v5-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-v5-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-v5-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF5/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
          dolor sit
        </h1>
      </div>
    </div>
  </section>
</pageheader>
`;

exports[`PageHeader Component should render the subtitle when viewId is provided: with subtitle 1`] = `
<pageheader
  props="{
  "productLabel": "RHEL",
  "children": [
    {}
  ]
}"
>
  <section
    class="pf-v5-l-page-header pf-v5-c-page-header pf-v5-l-page__main-section pf-v5-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-v5-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-v5-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-2"
          data-ouia-component-type="PF5/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
          lorem
        </h1>
      </div>
    </div>
    <p>
      t(curiosity-view.subtitle, {"appName":"Subscriptions","context":"RHEL"}, [object Object])
    </p>
  </section>
</pageheader>
`;
